### PR TITLE
Add `page_heading` option to text area

### PR DESCRIPTION
**Pull request by johnny-m-young**
## Changes
Adds the `page_heading` option to `Textarea`, `TextInput` and `DateInput` so that you can have the label as a page heading.

For the time being, the `page_heading` option overrides the `optional` option, meaning the `(optional)` part of the label is omitted.

## Rationale
I came across a few use cases for this when building the consumer forms in the Referrals application. I managed to work around it ([see here](https://github.com/citizensadvice/referrals/blob/7b79512576557e0b401647c061ae18a7b1b4b1f5/app/views/consumer/forms/post/steps/_desired_outcome.html.haml#L3)) but it felt a bit hacky. 

Further, the GOV.UK design system [suggests](https://design-system.service.gov.uk/get-started/labels-legends-headings/) nesting the `label` within `h1` tags rather than the other way around (which was the result of my solution).

### CODE REVIEW

**Comment by johnny-m-young** [comment]
Interested on your thoughts on this @davidrapson

File: contributing/testing.md [hunk 1]
```diff
@@ -10,6 +12,24 @@ From the project root run:
 npm run lint
 ```
 
+### Ruby linting
```
Comment (davidrapson) [inline contributing/testing.md:15]: Thanks for adding this! I have https://github.com/citizensadvice/design-system/pull/3496 open right now which makes some larger changes to these guides. I've tried to incorporate your changes into there. Mind taking a look and if you are happy pulling these changes out of this PR?

File: demo/backstop-common.js [hunk 2]
```diff
@@ -362,6 +362,11 @@ module.exports = function backstopCommon(baseUrl) {
         url: componentUrlFor(`textarea/value`),
         viewports: [{ label: 'form-field', width: 800, height: 400 }],
       },
+      {
```
Comment (davidrapson) [inline demo/backstop-common.js:365]: I'd love to see scenarios for each of the fields that have been updated

File: demo/spec/components/previews/textarea_preview.rb [hunk 3]
```diff
@@ -54,6 +54,18 @@ def error
...
   end
 
+  def page_heading
+    render(
+      CitizensAdviceComponents::Textarea.new(
+        name: "example-textarea-page-heading",
+        label: "Example text area with page heading",
+        options: {
+          page_heading: true
+        }
```
Comment (davidrapson) [inline demo/spec/components/previews/textarea_preview.rb:64]: In each of these examples do you mind if we include the `hint` option too for completeness?

**Review by davidrapson — CHANGES_REQUESTED** [review]
This is looking great! I think we'll want to extend this to cover all fields that inherit from `Input`, as the current change will affect any field that uses it. This would be `TextInput`, `DateInput` and (the one you've already covered) `Textarea`.  If you do only want to scope this to the `Textarea` we'll need to narrow the scope of the changes somewhat. Arguably we should include fields like radio and checkbox collections here too but I think that's probably too much for this PR.

I'd also suggest chatting with @leeky on how to integrate this into https://github.com/citizensadvice/rails-form-builder as I think for referrals you ultimately want these changes there too.

For your questions:

1. Yes, the order of the heading and label wrapping makes a big difference. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#headings (will see if I can dig out a more detailed summary)
2. I think we can probably make the call to omit the optional label when the field uses a page heading. We'd need to give that a bit more thought and some extra styles anyway so probably fine to side-step for now.

Great work on this and thanks so much for the contribution!

File: contributing/testing.md (hunk 1 above)
Comment (johnny-m-young) [reply contributing/testing.md:15]: Sorry just catching up after holidays. #3496 looks good, happy to take these out

File: engine/spec/components/citizens_advice_components/input_spec.rb [hunk 4]
```diff
@@ -72,6 +72,19 @@
...
+    before do
+      render_inline described_class.new(
+        name: "example-input-page-heading",
+        label: "Example input with page heading",
+        type: :text,
+        options: { page_heading: true }
+      )
+    end
+
+    it { is_expected.to have_css "h1[class=cads-page-title]", text: "Example input with page heading" }
```
Comment (davidrapson) [inline engine/spec/components/citizens_advice_components/input_spec.rb:85]: This takes a CSS selector so it can be just `h1.cads-page-title`

**Review by davidrapson — APPROVED** [review]
Lovely work on this 🥳, we'll still need to add this to  https://github.com/citizensadvice/rails-form-builder once this is out.

Once last request before merging, can you add something to the CHANGELOG?

**Pull request merged**